pkg/controller/controller.go

```diff
@@ -126,6 +130,19 @@ func NewUnmanaged(name string, mgr manager.Manager, options Options) (Controller
 		return nil, err
 	}
 
+	// Add controller and reconciler group / kind to logger.
+	log := options.Log.WithValues("controller", name)
+	if gvk != nil {
+		log = log.WithValues("reconciler group", gvk.Group, "reconciler kind", gvk.Kind)
+	}
+
+	// Initialize random source, later used to generate reconcileIDs.
+	var rngSeed int64
+	if err := binary.Read(crand.Reader, binary.LittleEndian, &rngSeed); err != nil {
+		return nil, fmt.Errorf("could not read random bytes to seed random source for reconcileID generation: %v", err)
+	}
+	randSource := rand.New(rand.NewSource(rngSeed)) //nolint:gosec // math/rand is enough, we don't need crypto/rand.
```

Comment:
Is there any reason not to use crypto/rand anyway?

Reply:
Would be fine for me. I essentially copied this from the traceID/spanID generator from opentelemetry

Maybe it needs less "randomness"? :)

Reply:
Based on https://blog.gopheracademy.com/advent-2017/a-tale-of-two-rands/ I would prefer using math rand. crypto rand seems to be way slower. I think for us it's good enough to seed math rand with crypto and then just use the pseudo number generator provided by math rand.

Reply:
Yeah I think math rand should be enough as well, we don't expect cryptographic  security in these IDs.

But why not use `math/rand.Read` or maybe even `github.com/google/uuid`?

Reply:
I'll check. I was mostly just reusing what they are doing in open telemetry to end up with the same format as traceIDs. But I don't mind diverging there is no strict reason why we have to use the same.

Reply:
Let's use k8s.io/apimachinery/pkg/util/uuid (which is using the google lib under the hood)